frame/aura/src/lib.rs

```diff
@@ -179,8 +187,12 @@ impl<T: Config> OneSessionHandler<T::AccountId> for Pallet<T> {
 		if changed {
 			let next_authorities = validators.map(|(_, k)| k).collect::<Vec<_>>();
 			let last_authorities = Self::authorities();
-			if next_authorities != last_authorities {
-				Self::change_authorities(next_authorities);
+			if last_authorities != next_authorities {
+				let bounded = match next_authorities.try_into() {
+					Ok(v) => v,
+					Err(_) => return,
+				};
+				Self::change_authorities(bounded);
```

Comment:
I haven't read deeply, but should we make the users of `on_new_session` check to see if the `validators` param respects `MaxAuthorities? Or have a proof here that it is ok to simply not use the new validators if there are too many?

Reply:
If it ever overflow current behaviour is simply do nothing and return early (as per the match expression above). I'm currently looking at an old PR to figure out what the best way forward is.